libyul/Object.cpp

```diff
@@ -128,18 +128,33 @@ vector<size_t> Object::pathToSubObject(YulString _qualifiedName) const
 
 	vector<size_t> path;
 	Object const* object = this;
-	for (string const& currentSubObjectName: subObjectPathComponents)
+	for (size_t i = 0; i < subObjectPathComponents.size(); i++)
 	{
+		auto const& currentSubObjectName = subObjectPathComponents[i];
 		yulAssert(!currentSubObjectName.empty(), "");
 		auto subIndexIt = object->subIndexByName.find(YulString{currentSubObjectName});
 		yulAssert(
 			subIndexIt != object->subIndexByName.end(),
-			"Assembly object <" + _qualifiedName.str() + "> not found or does not contain code."
+			"Assembly object <" + _qualifiedName.str() + "> not found."
 		);
-		object = dynamic_cast<Object const*>(object->subObjects[subIndexIt->second].get());
-		yulAssert(object, "Assembly object <" + _qualifiedName.str() + "> not found or does not contain code.");
-		yulAssert(object->subId != numeric_limits<size_t>::max(), "");
-		path.push_back({object->subId});
+		auto const* objectNode = object->subObjects[subIndexIt->second].get();
+		object = dynamic_cast<Object const*>(objectNode);
+		if (object)
+		{
+			yulAssert(object->subId != numeric_limits<size_t>::max());
+			path.emplace_back(object->subId);
+		}
+		else
+		{
+			yulAssert(
+				i == subObjectPathComponents.size() - 1,
+				"Assembly object <" + _qualifiedName.str() + "> of incorrect path."
```

Comment:
This would be more accurate:

```suggestion
				"Assembly object <" + _qualifiedName.str() + "> does not exist. <" + path + "> has no nested objects."
```

But you'd have to use something like `joinHumanReadable()` on path to convert it to a string.